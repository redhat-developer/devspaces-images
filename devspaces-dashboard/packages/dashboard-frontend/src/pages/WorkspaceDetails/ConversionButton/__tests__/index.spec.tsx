/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render, RenderResult, screen, waitFor } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import WorkspaceConversionButton from '..';
import userEvent from '@testing-library/user-event';

const mockOnConvert = jest.fn();

describe('WorkspaceConversionButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('render', () => {
    const component = getComponent();
    const snapshot = createSnapshot(component);
    expect(snapshot).toMatchSnapshot();
  });

  it('should emit cleanupError on conversion start', async () => {
    const component = getComponent();
    renderComponent(component);

    const convertButton = screen.getByRole('button', { name: 'Convert' });
    userEvent.click(convertButton);

    await waitFor(() => expect(mockOnConvert).toBeCalled());
  });

  function getComponent(): React.ReactElement {
    return <WorkspaceConversionButton onConvert={mockOnConvert} />;
  }

  function renderComponent(component: React.ReactElement): RenderResult {
    return render(component);
  }

  function createSnapshot(
    component: React.ReactElement,
  ): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    return renderer.create(component).toJSON();
  }
});
