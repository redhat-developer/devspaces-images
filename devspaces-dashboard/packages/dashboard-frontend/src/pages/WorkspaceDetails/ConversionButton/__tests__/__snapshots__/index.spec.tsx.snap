// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceConversionButton render 1`] = `
<button
  aria-disabled={false}
  aria-label={null}
  className="pf-c-button pf-m-warning"
  data-ouia-component-id="OUIA-Generated-Button-warning-1"
  data-ouia-component-type="PF4/Button"
  data-ouia-safe={true}
  disabled={false}
  onClick={[Function]}
  role={null}
  type="button"
>
  Convert
</button>
`;
